zerver/worker/queue_processors.py

```diff
@@ -391,3 +393,24 @@ def consume(self, event):
         logging.info("TestWorker should append this message to %s: %s" % (fn, message))
         with open(fn, 'a') as f:
             f.write(message + '\n')
+
+@assign_queue('embed_links')
+class FetchLinksEmbedData(QueueProcessingWorker):
+    def consume(self, event):
+        # type: (Mapping[str, Any]) -> None
+        for url in event['urls']:
+            url_preview.get_link_embed_data(url)
+
+        message = Message.objects.get(id=event['message_id'])
+        if message.content != event['message_content']:
+            return
+        if message.content is not None:
+            ums = UserMessage.objects.filter(message=message.id)
+            message_users = {get_user_profile_by_id(um.user_profile_id) for um in ums}
```

Comment:
This will do O(N) database queries and perform very badly; you should instead include a `.select_related("user_profile")` sort of thing on the `UserMessage` query, and then just have this line be `[um.user_profile for um in ums]`.